artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/wireformat/ReplicationPageEventMessage.java

```diff
@@ -54,7 +54,7 @@ public int expectedEncodeSize() {
    @Override
    public void encodeRest(final ActiveMQBuffer buffer) {
       buffer.writeSimpleString(storeName);
-      buffer.writeInt(pageNumber);
+      buffer.writeInt((int)pageNumber);
```

Comment:
seems like it should at least validate the number before truncating it...and then perhaps explode if it would have done the wrong thing?

Reply:
sh***t.... I forgot about this...


I will need some versioning work here.

Thanks...

Reply:
I added versioning on the protocol, and a compatibility test where I'm replicating between servers on different versions and using paging with multiple protocols, and it's all good.

there's still a small possibility of a trunk if replication to or from a previous version, but that will happen only if the user keeps running it forever like this..  I don't think this is something that needs to be addressed.